### ISSUE TITLE
Importing private virtual circuit fails because of public_prefixes count change

### ISSUE THREAD

**Issue by lucianf** [terraform-providers/terraform-provider-oci#700]
I'm trying to import an existing FastConnect virtual circuit into an `oci_core_virtual_circuit` resource.  I've created the resource in Terraform and imported the existing virtual circuid OCID.

`terraform plan` informs that a new resource is required:

> public_prefixes.#:                                                "" => <computed> (forces new resource)

However this is a private virtual circuits, and `public_prefixes` does not exist in the resource configuration.  Therefore recomputing it should again yield null, and should not force a new resource - so I believe this is a bug.

Or am I missing something?

### Terraform Version

<details>
Terraform v0.11.11
+ provider.oci v3.13.0
</details>

### OCI Provider Version

<details>
2019/01/28 14:57:28 [INFO] terraform-provider-oci 3.13.0
</details>

### Terraform Resource

<details>

```
resource "oci_core_virtual_circuit" "ocidata_prod_virtual_circuit" {
  compartment_id       = "${oci_identity_compartment.xx.id}"
  type                 = "PRIVATE"
  bandwidth_shape_name = "1 Gbps"

  cross_connect_mappings {
    cross_connect_or_cross_connect_group_id = "ocid1.crossconnectgroup.oc1.xxx"
    customer_bgp_peering_ip                 = "xxxx/31"
    oracle_bgp_peering_ip                   = "yyyy/31"
    vlan                                    = nnnn
  }

  customer_bgp_asn    = nnnnnn
  display_name        = "xxxxxxx"
  gateway_id          = "ocid1.drg.oc1.xxx"
  provider_service_id = "ocid1.providerservice.oc1.xxx"
}
```

</details>

### Terraform Plan

<details>

```
-/+ oci_core_virtual_circuit.ocidata_prod_virtual_circuit (new resource required)
      id:                                                               "ocid1.virtualcircuit.xxx" => <computed> (forces new resource)
      bandwidth_shape_name:                                             "1 Gbps" => "1 Gbps"
      bgp_management:                                                   "CUSTOMER_MANAGED" => <computed>
      bgp_session_state:                                                "UP" => <computed>
      compartment_id:                                                   "ocid1.compartment.oc1..xxx" => "ocid1.compartment.oc1..xxx"
      cross_connect_mappings.#:                                         "1" => "1"
      cross_connect_mappings.0.bgp_md5auth_key:                         "" => <computed>
      cross_connect_mappings.0.cross_connect_or_cross_connect_group_id: "ocid1.crossconnectgroup.oc1.xxx" => "ocid1.crossconnectgroup.oc1.xxx"
      cross_connect_mappings.0.customer_bgp_peering_ip:                 "xxxx/31" => "xxxx/31"
      cross_connect_mappings.0.oracle_bgp_peering_ip:                   "yyyy/31" => "yyyy/31"
      cross_connect_mappings.0.vlan:                                    "mmmm" => "nnnn"
      customer_bgp_asn:                                                 "nnnnnn" => "nnnnnn"
      display_name:                                                     "xxxxxxx" => "xxxxxxx"
      gateway_id:                                                       "ocid1.drg.oc1.xxx" => "ocid1.drg.oc1.xxx"
      oracle_bgp_asn:                                                   "31898" => <computed>
      provider_service_id:                                              "ocid1.providerservice.oc1.xxx" => "ocid1.providerservice.oc1.xxx"
      provider_state:                                                   "ACTIVE" => <computed>
      public_prefixes.#:                                                "" => <computed> (forces new resource)
      reference_comment:                                                "" => <computed>
      region:                                                           "" => <computed>
      service_type:                                                     "LAYER2" => <computed>
      state:                                                            "PROVISIONED" => <computed>
      time_created:                                                     "2019-01-25 16:11:22.773 +0000 UTC" => <computed>
      type:                                                             "PRIVATE" => "PRIVATE"


Plan: 1 to add, 0 to change, 1 to destroy.
```

</details>

**Comment by afedorch** [comment]
Thanks for reporting the issue. 
As a current workaround you can mark the `public_prefixes` as ignored attribute by updating the resource config:
```
  lifecycle {
    ignore_changes = ["public_prefixes"]
  }
```

**Comment by ravitandonrt** [comment]
Fix for this is available with version v3.14.1
